src/constraints.jl

```diff
@@ -482,7 +482,8 @@ function VectorConstraint(
     func::AbstractVector{<:AbstractJuMPScalar},
     set::MOI.AbstractVectorSet,
 )
-    return VectorConstraint(collect(func), set)
+    f = [func[idx] for idx in eachindex(func)]
```

Comment:
Maybe just add a comment above this line saying that we don't use `collect` so that it work for DenseAxisArray